integration/tests/cook/util.py

```diff
@@ -811,16 +810,26 @@ def load_instance(cook_url, instance_uuid, assert_response=True):
     return load_resource(cook_url, 'instances', instance_uuid, assert_response)
 
 
-def wait_until(query, predicate, max_wait_ms=DEFAULT_TIMEOUT_MS, wait_interval_ms=DEFAULT_WAIT_INTERVAL_MS):
+def wait_until(query, predicate, max_wait_ms=DEFAULT_TIMEOUT_MS, wait_interval_ms=DEFAULT_WAIT_INTERVAL_MS, fail_on_error=False):
     """
     Block until the predicate is true for the result of the provided query.
     `query` is a thunk (nullary callable) that may be called multiple times.
     `predicate` is a unary callable that takes the result value of `query`
     and returns True if the condition is met, or False otherwise.
+
+    This will retry until timedout or until a non-RuntimeError is thrown 
```

Comment:
This comment is a little hard to parse, can you rephrase it to something like "if `fail_on_error` is true and a non-RuntimeError is thrown, the method will stop retrying"